iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/vector/HiveBatchIterator.java

```diff
@@ -144,4 +115,53 @@ public HiveBatchContext next() {
     advanced = false;
     return new HiveBatchContext(batch, vrbCtx, rowOffset);
   }
+
+  public void fillVirtualColumns() {
```

Comment:
keep it in `advance` method

Reply:
Had to do this because of checkstyle.

Reply:
hm, then why it didn't complain before? you just added extra case
shouldn't it be `private` then? also maybe just `setVirtualColumns`